### PR TITLE
[Fix](fe) Avoid interrupt daemon thread and use proper polling interval

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

use a proper sleep interval to balance both:
a. no interrupt so no bdbje writing break which leads to crash
b. the daemon threads don't need much time to know its interval change.

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/common/util/Daemon.java [hunk 1]
```diff
@@ -119,9 +119,23 @@ public void run() {
...
+                    Thread.sleep(5000L);
+                    remainingInterval -= 5000L;
+                }
+                if (remainingInterval <= 5000L) {
+                    Thread.sleep(remainingInterval);
+                }
             } catch (InterruptedException e) {
-                LOG.info("InterruptedException: ", e);
+                // the daemon thread should NOT be interrupted. or meet bdbje writing, it will be disaster.
+                LOG.warn("InterruptedException: ", e);
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/Daemon.java:138]: disaster only log warning not fatal?

**Comment by zclllyybb** [comment]
run buildall

**Review by swjtu-zhanglei — APPROVED** [review]

**Review by zhangstar333 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java [hunk 2]
```diff
@@ -806,6 +807,48 @@ private void initDynamicPartitionTable() {
...
+            try {
+                runOneCycle();
+            } catch (Throwable e) {
+                LOG.error("daemon thread got exception. name: {}", getName(), e);
+            }
+
+            try {
+                long oldInterval = intervalMs;
+                long remainingInterval = oldInterval;
+                while (remainingInterval > 5000L) {
```
Comment (BiteTheDDDDt) [inline fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java:828]: avoid magic number 5000

**Comment by zclllyybb** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**